core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -193,12 +194,11 @@ private[spark] class DiskBlockObjectWriter(
     }
 
     objOut.writeObject(value)
+    numRecordsWritten += 1
```

Comment:
What about just adding a class level comment that it can't be used after it is closed?